From 333fd70c3d70d58ebd4737e49a76c14da627a5b2 Mon Sep 17 00:00:00 2001
From: Rong Bao <webmaster@csmantle.top>
Date: Thu, 27 Aug 2026 02:16:25 +0000
Subject: [PATCH 08/19] Bug 2066961 - Enable -fstack-clash-protection on
 loongarch64 Clang>=23.1.0. r=glandium!

This feature was merged into Clang's main tree on May 28, 2026 and was
released with Clang 23.1.0.

Link: https://github.com/llvm/llvm-project/pull/195595 ("[LoongArch] Add
 -fstack-clash-protection support")

Differential Revision: https://phabricator.services.mozilla.com/D321754
---
 build/moz.configure/toolchain.configure | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 4a416b5476f802c6322271a5229e0c97867055c4..82c1c9c8055fb0a8b0d439056124d4d15e08fa03 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -2726,7 +2726,9 @@ def security_hardening_cflags(
                     or
                      (c_compiler.version >= "18.1.0" and target.cpu == "aarch64")
                     or
-                     (c_compiler.version >= "22.1.8" and target.cpu == "riscv64"))
+                     (c_compiler.version >= "22.1.8" and target.cpu == "riscv64")
+                    or
+                     (c_compiler.version >= "23.1.0" and target.cpu == "loongarch64"))
             ):
                 flags.append("-fstack-clash-protection")
                 ldflags.append("-fstack-clash-protection")
-- 
2.55.0

